### PR TITLE
Hide player config when it’s locked by server

### PR DESCRIPTION
**Pull request by AbandonedCart**
## Summary by Sourcery

Add server-side control to lock auto-skip settings in the Intro Skipper plugin, preventing client-side changes when the server enforces specific configurations. Extend the configuration class to support new auto-skip properties and client-specific settings.

New Features:
- Introduce the ability to lock auto-skip functionality based on server configuration, preventing users from changing auto-skip settings when enforced by the server.

Enhancements:
- Extend the UserInterfaceConfiguration class to include new properties for auto-skip settings and client-specific configurations.

### CODE REVIEW

**Comment by AbandonedCart** [comment]
This hurts my brain. I forgot to retrieve it

**Comment by AbandonedCart** [comment]
That’s 90% **unlikely** to work

**Comment by jumoog** [comment]
i'll take the other 10%

**Comment by AbandonedCart** [comment]
@sourcery-ai review

File: ConfusedPolarBear.Plugin.IntroSkipper/Configuration/inject.js [hunk 1]
```diff
@@ -303,8 +303,19 @@ const introSkipper = {
...
     },
+    isAutoSkipLocked(config) {
+        const isAutoSkip = config.AutoSkip && config.AutoSkipCredits;
+        const isAutoSkipClient = config.SkipButtonVisible && 
+            config.ClientList.split(",").includes(ApiClient.appName());
+        return isAutoSkip || isAutoSkipClient;
+    },
     injectIntroSkipperOptions(actionSheet) {
         if (!this.skipButton) return;
+        const config = (async () => await this.secureFetch("Intros/UserInterfaceConfiguration"))();
```
[the remark this answers is not in the document]
Comment (AbandonedCart) [reply ConfusedPolarBear.Plugin.IntroSkipper/Configuration/inject.js:314]: There we go

[the remark this answers is not in the document]
Comment (AbandonedCart) [reply ConfusedPolarBear.Plugin.IntroSkipper/Configuration/inject.js:314]: @sourcery-ai suggest a change

**Comment by AbandonedCart** [comment]
@sourcery-ai review

File: ConfusedPolarBear.Plugin.IntroSkipper/Configuration/inject.js [hunk 2]
```diff
@@ -303,8 +303,19 @@ const introSkipper = {
...
+    isAutoSkipLocked(config) {
+        const isAutoSkip = config.AutoSkip && config.AutoSkipCredits;
+        const isAutoSkipClient = config.SkipButtonVisible && 
+            config.ClientList.split(",").includes(ApiClient.appName());
+        return isAutoSkip || isAutoSkipClient;
+    },
+    async injectIntroSkipperOptions(actionSheet) {
         if (!this.skipButton) return;
+        const config = await this.secureFetch("Intros/UserInterfaceConfiguration");
+        if (isAutoSkipLocked(config)) {
```
[the remark this answers is not in the document]
Comment (AbandonedCart) [reply ConfusedPolarBear.Plugin.IntroSkipper/Configuration/inject.js:315]: Good catch

**Comment by AbandonedCart** [comment]
So now it worries about performance? Cool…

**Comment by AbandonedCart** [comment]
I still have mixed feelings on if this is a rabbit hole we really want to jump down.

**Pull request merged**